### PR TITLE
Expose maat_reference on defendant

### PR DESCRIPTION
**Pull request by mpw5**
## What

Fixes https://github.com/ministryofjustice/laa-court-data-adaptor/issues/113

Replace the `is_linked` attribute on the defendant model with `maat_reference`. This returns the `maat_reference` of the defendant's first offence unless the `maat_reference` begins with 'Z' - this indicates that the case has been unlinked, and so `nil` is returned instead.

`is_linked` is no longer required as View Court Data can infer this from the presence (or not) of a `maat_reference`.

## Checklist

Before you ask people to review this PR:

- [x] Tests and linters should be passing
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Comment by mpw5** [comment]
Thanks @archferns. Good point. I've updated `swagger/v1/defendant.json` too.

File: app/models/defendant.rb [hunk 1]
```diff
@@ -37,7 +37,9 @@ def offence_ids
     offences.map(&:id)
   end
 
-  def linked?
-    offences.map(&:maat_reference).compact.present?
+  def maat_reference
+    return nil if offences == [] || offences.first.maat_reference&.first == 'Z'
```
Comment (jsugarman) [inline app/models/defendant.rb]:
```suggestion
    return nil if offences.empty? || offences.first.maat_reference&.first&.eql?( 'Z')
```

Comment (jsugarman) [reply app/models/defendant.rb]: is it ever possible to have different `maat_references` on offences?

perhaps there should be something like below. and if it ever returns more that one then an error should be raise
```
offences.map(&:maat_reference).uniq.compact
```

Comment (jsugarman) [reply app/models/defendant.rb]: **^^ not a blocker**

File: spec/models/defendant_spec.rb [hunk 2]
```diff
@@ -21,7 +21,20 @@
...
+    let(:offences) do
+      [instance_double('Offence', maat_reference: nil),
+       instance_double('Offence', maat_reference: nil)]
+    end
+
+    before do
+      allow(defendant).to receive(:offences).and_return(offences)
+    end
+
+    it { expect(defendant.maat_reference).to eq(nil) }
```
Comment (jsugarman) [inline spec/models/defendant_spec.rb]:
```suggestion
    it { expect(defendant.maat_reference).to be_nil }
```
https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/be-matchers

File: app/models/defendant.rb (hunk 1 above)
Comment (jsugarman) [reply app/models/defendant.rb]: just to mention that `string.first` is a [rails extension](https://api.rubyonrails.org/classes/String.html#method-i-first).
consider alternative `start_with?` ruby string class method which may be clearer:
```
offences.first.maat_reference.start_with?('Z')
```

Comment (jsugarman) [reply app/models/defendant.rb]: have added a commit to handle these @mpw5

**Pull request merged**